### PR TITLE
RS: TLS 1.3

### PR DESCRIPTION
**Pull request by rrelledge**
DOC-2445

Staged previews:
- [Configure TLS protocol](https://docs.redis.com/staging/DOC-2445/rs/security/encryption/tls/tls-protocols/) - will need to replace the screenshot after the radio button label is updated
- [Configure cipher suites](https://docs.redis.com/staging/DOC-2445/rs/security/encryption/tls/ciphers/)
- [rladmin cluster config](https://docs.redis.com/staging/DOC-2445/rs/references/cli-utilities/rladmin/cluster/config/)
- [Cluster REST API object](https://docs.redis.com/staging/DOC-2445/rs/references/rest-api/objects/cluster/)
- [Internode encryption > Encryption ciphers and settings](https://docs.redis.com/staging/DOC-2445/rs/security/encryption/internode-encryption/#encryption-ciphers-and-settings)

### CODE REVIEW

File: content/rs/references/rest-api/objects/cluster/_index.md [hunk 1]
```diff
@@ -47,8 +49,9 @@ An API object that represents the cluster.
 | reserved_ports | array of strings | List of reserved ports and/or port ranges to avoid using for database endpoints (for example <nobr>`"reserved_ports": ["11000", "13000-13010"]`</nobr>) |
 | s3_url | string | Specifies the URL for S3 export and import |
 | saslauthd_ldap_conf | string | saslauthd LDAP configuration |
-| sentinel_cipher_suites | array | Specifies the list of enabled ciphers for the sentinel service. The supported ciphers are ones that were implemented by the [cipher_suits.go](<https://golang.org/src/crypto/tls/cipher_suites.go>) package. |
-| sentinel_ssl_policy | 'allowed'<br />'disabled' <br />'required' | Determines whether the discovery service allows, blocks, or requires TLS connections<br />**allowed**: Allows both TLS and non-TLS connections<br />**disabled**: Allows only non-TLS connections<br />**required**: Allows only TLS connections |
+| sentinel_cipher_suites | array | Specifies the list of enabled ciphers for the sentinel service. The supported ciphers are ones that were implemented by the [cipher_suites.go](<https://golang.org/src/crypto/tls/cipher_suites.go>) package. |
```
Comment (mich-elle-luna) [inline content/rs/references/rest-api/objects/cluster/_index.md:52]:
```suggestion
| sentinel_cipher_suites | array | Specifies the list of enabled ciphers for the sentinel service. The supported ciphers are those implemented by the [cipher_suites.go](<https://golang.org/src/crypto/tls/cipher_suites.go>) package. |
```

File: content/rs/references/rest-api/objects/cluster/_index.md [hunk 2]
```diff
@@ -47,8 +49,9 @@ An API object that represents the cluster.
...
+| sentinel_cipher_suites_tls_1_3 | string | Specifies the list of enabled TLS 1.3 ciphers for the discovery (sentinel) service. The supported ciphers are ones that were implemented by the [cipher_suites.go](<https://golang.org/src/crypto/tls/cipher_suites.go>) package. |
```
Comment (mich-elle-luna) [inline content/rs/references/rest-api/objects/cluster/_index.md:53]:
```suggestion
| sentinel_cipher_suites_tls_1_3 | string | Specifies the list of enabled TLS 1.3 ciphers for the discovery (sentinel) service. The supported ciphers are those implemented by the [cipher_suites.go](<https://golang.org/src/crypto/tls/cipher_suites.go>) package. |
```

**Review by mich-elle-luna — COMMENTED** [review]
Just a comment for those instead of ones because it is a bit awkward to read. Looks good otherwise!

File: static/images/rs/screenshots/cluster/security-tls-protocols-edit.png
Comment (yoavredis) [inline static/images/rs/screenshots/cluster/security-tls-protocols-edit.png:1]: We need to replace the image with the corrected title (the duplicated one).

File: content/rs/security/encryption/tls/tls-protocols.md [hunk 3]
```diff
@@ -18,49 +18,67 @@ The communications for which you can modify TLS protocols are:
...
 If you have trouble enabling specific versions of TLS, verify that they're supported by your operating system and that they're configured correctly.
 
 {{<note>}}
 TLSv1.2 is generally recommended as the minimum TLS version for encrypted communications. Check with your security team to confirm which TLS protocols meet your organization's policies.
 {{</note>}}
 
-### Control plane
+### Edit TLS settings in the UI {#edit-tls-ui}
+
+To configure cipher suites using the Cluster Manager UI:
```
Comment (yoavredis) [inline content/rs/security/encryption/tls/tls-protocols.md:39]: This section is not for configuring cipher suites, but for configuring the minimum TLS versions

File: content/rs/references/cli-utilities/rladmin/cluster/config.md [hunk 4]
```diff
@@ -54,12 +57,14 @@ Updates the cluster configuration.
 | audit_address | string | TCP/IP address where a listener can capture [audit event notifications]({{<relref "/rs/security/audit-events">}}) |
 | audit_port | string | Port where a listener can capture [audit event notifications]({{<relref "/rs/security/audit-events">}}) |
 | audit_protocol | `tcp`<br/>`local` | Protocol used for [audit event notifications]({{<relref "/rs/security/audit-events">}})<br/>For production systems, only `tcp` is supported. |
-| cipher_suites | list of ciphers | Cipher suites used for TLS connections to the admin console (specified in the format understood by the BoringSSL library) |
+| control_cipher_suites | list of ciphers | Cipher suites used for TLS connections to the admin console (specified in the format understood by the BoringSSL library)<br />(previously named `cipher_suites`) |
+| control_cipher_suites_tls_1_3 | list of ciphers |  Specifies the enabled TLS 1.3 ciphers for the control plane. The ciphers are specified in the format understood by the BoringSSL library. |
```
Comment (yoavredis) [inline content/rs/references/cli-utilities/rladmin/cluster/config.md:61]: control_cipher_suites_tls_1_3 - add that it's read only.

File: content/rs/references/cli-utilities/rladmin/cluster/config.md [hunk 5]
```diff
@@ -69,14 +74,15 @@ Updates the cluster configuration.
...
 | reserved_ports | list of ports/port ranges | List of reserved ports and/or port ranges to avoid using for database endpoints (for example `reserved_ports 11000 13000-13010`) |
 | s3_url | string | The URL of S3 export and import |
 | saslauthd_ldap_conf | filepath | Updates LDAP authentication configuration for the cluster |
-| sentinel_cipher_suites | list of ciphers | Cipher suites used by the discovery service (supported ciphers are implemented by the [golang.org cipher suites package](https://golang.org/src/crypto/tls/cipher_suites.go)) |
-| sentinel_ssl_policy | `allowed`<br />`required`<br />`disabled` | Define the SSL policy for the discovery service |
+| sentinel_cipher_suites | list of ciphers | Cipher suites used by the discovery service (supported ciphers are implemented by the [cipher_suites.go](<https://golang.org/src/crypto/tls/cipher_suites.go>) package) |
+| sentinel_cipher_suites_tls_1_3 | list of ciphers | Specifies the list of enabled TLS 1.3 ciphers for the discovery (sentinel) service. The supported ciphers are ones that were implemented by the [cipher_suites.go](<https://golang.org/src/crypto/tls/cipher_suites.go>) package. |
```
Comment (yoavredis) [inline content/rs/references/cli-utilities/rladmin/cluster/config.md:84]: sentinel_cipher_suites_tls_1_3 - add that it's read only

File: content/rs/references/rest-api/objects/cluster/_index.md [hunk 6]
```diff
@@ -19,10 +19,12 @@ An API object that represents the cluster.
 | cnm_http_port | integer, (range:&nbsp;1024-65535) | API HTTP listening port |
 | cnm_https_port | integer, (range:&nbsp;1024-65535) | API HTTPS listening port |
 | control_cipher_suites | string | Specifies the enabled ciphers for the control plane. The ciphers are specified in the format understood by the BoringSSL library. |
+| control_cipher_suites_tls_1_3 | string | Specifies the enabled TLS 1.3 ciphers for the control plane. The ciphers are specified in the format understood by the BoringSSL library. |
```
Comment (yoavredis) [inline content/rs/references/rest-api/objects/cluster/_index.md:22]: control_cipher_suites_tls_1_3 - add that it's read only

File: content/rs/references/rest-api/objects/cluster/_index.md [hunk 7]
```diff
@@ -47,8 +49,9 @@ An API object that represents the cluster.
...
 | s3_url | string | Specifies the URL for S3 export and import |
 | saslauthd_ldap_conf | string | saslauthd LDAP configuration |
-| sentinel_cipher_suites | array | Specifies the list of enabled ciphers for the sentinel service. The supported ciphers are ones that were implemented by the [cipher_suits.go](<https://golang.org/src/crypto/tls/cipher_suites.go>) package. |
-| sentinel_ssl_policy | 'allowed'<br />'disabled' <br />'required' | Determines whether the discovery service allows, blocks, or requires TLS connections<br />**allowed**: Allows both TLS and non-TLS connections<br />**disabled**: Allows only non-TLS connections<br />**required**: Allows only TLS connections |
+| sentinel_cipher_suites | array | Specifies the list of enabled ciphers for the sentinel service. The supported ciphers are those implemented by the [cipher_suites.go](<https://golang.org/src/crypto/tls/cipher_suites.go>) package. |
+| sentinel_cipher_suites_tls_1_3 | string | Specifies the list of enabled TLS 1.3 ciphers for the discovery (sentinel) service. The supported ciphers are those implemented by the [cipher_suites.go](<https://golang.org/src/crypto/tls/cipher_suites.go>) package. |
```
Comment (yoavredis) [inline content/rs/references/rest-api/objects/cluster/_index.md:53]: sentinel_cipher_suites_tls_1_3 - add that it's read only

File: content/rs/security/encryption/internode-encryption.md
Comment (yoavredis) [inline content/rs/security/encryption/internode-encryption.md:1]: I think that this line has a mistake:
"To enable internode encryption for a database (also called _data plane encryption_)"

I believe it should be "(also called _data internode encription_)"

**Review by yoavredis — CHANGES_REQUESTED** [review]
Left a few comments.
10x

File: content/rs/security/encryption/tls/tls-protocols.md (hunk 3 above)
Comment (rrelledge) [reply content/rs/security/encryption/tls/tls-protocols.md:39]: Oops. Nice catch. I will fix this.

**Review by kaitlynmichael — APPROVED** [review]

File: static/images/rs/screenshots/cluster/security-tls-protocols-edit.png
Comment (rrelledge) [reply static/images/rs/screenshots/cluster/security-tls-protocols-edit.png:1]: Updated screenshot

File: static/images/rs/screenshots/cluster/security-tls-cipher-suites-edit.png
[the remark this answers is not in the document]
Comment (rrelledge) [reply static/images/rs/screenshots/cluster/security-tls-cipher-suites-edit.png:1]: Updated screenshot

File: content/rs/security/encryption/internode-encryption.md
Comment (rrelledge) [reply content/rs/security/encryption/internode-encryption.md:1]: Fixed

**Pull request merged**